src/assignments/orbit-report.rst

```diff
@@ -0,0 +1,414 @@
+
+#. Add this HTML ``<div class="search-form"></div>`` in your ``app.component.html``.
+#. Add an ``<input>`` element insde the ``<div>``.
+#. Add a ``<button>`` element insde the ``<div>``.
+#. Add an Angular click handler to the ``<button>`` that when clicked calls ``search(searchTerm.value)``
+
+   * ``searchTerm`` being the local variable name for the ``<input>``
+
+#. Add an Angular keyup.enter handler to the ``<input>`` that when clicked calls ``search(searchTerm.value)``
+
+   * ``searchTerm`` being the local variable name for the ``<input>``
+
+#. Add a ``search`` method to the ``AppComponent`` class.
+
+   * The ``search`` method is provded below.
+
+.. sourcecode:: typescript
+   :linenos:
+
+   search(searchTerm: string): void {
+      let matchingSatellites: Satellite[] = [];
+      searchTerm = searchTerm.toLowerCase();
+      for(let i=0; i < this.sourceList.length; i++) {
+         let name = this.sourceList[i].name.toLowerCase();
+         if (name.indexOf(searchTerm) >= 0) {
+            matchingSatellites.push(this.sourceList[i]);
+         }
+      }
+      // assign this.displayList to be the the array of matching satellites
+      // this will cause Angular to re-make the table, but now only containg matches
+      this.displayList = matchingSatellites;
+   }
+
+Notice the usage of a new variable named ``displayList``. ``displayList`` should contain the ``Satellite`` objects
+that the user wants to see. That was previously ALL the satellites. Now the user can perform a search, which
+means they want to see the matching results. ``sourceList`` will contain ALL the ``Satellite`` objects. If you removed
+the ``Satellite`` objects from ``sourceList``, then the user could never see them again. Instead when the user
+performs a search, ``displayList`` will be populated with the all the matching ``Satellite`` objects in ``sourceList``.
+Matching is defined as ``satellite.name`` containing the search term.
+
+7. Add the ``displayList: Satellite[];`` property to the ``AppComponent`` class.
+
+   * Set ``displayList = []`` in the constructor.
+
+8. Pass in the ``displayList`` to the ``orbit-list-component``.
+
+   * ``<app-orbit-list [satellites]="displayList"></app-orbit-list>``
+
+9. For an example of search working, see :ref:`Orbit Report Demo <orbit-report-demo>`.
+
+
+9) Counting Satellites
+^^^^^^^^^^^^^^^^^^^^^^
+For the last feature, you are on your own. You are tasked with creating a new component that
+show the total count and count by type for the satellites currently displayed in the table.
+
+#. Create an ``orbit-counts`` component.
+#. Copy the entire contents of the `example orbit-counts.component.css  <https://gist.github.com/welzie/5247f5ac36e973903cd5202af50932e6>`_ and put it into your ``orbit-counts.component.css``.
+#. Use the component in ``app.component.html``.
+#. Pass in ``displayList`` via ``[satellites]="displayList"``.
+#. Use the given HTML as a template.
```

Comment:
Where does this go?

Reply:
It goes in the new components template file `orbit-counts.component.html`.  Fewer instructions were given here to make the student figure these out on their own.

Reply:
I figured it out, but I think my comment above better describes my issue with this section.

Reply:
Again I appreciate the suggestion, but I really want to make the last section a "do it on your own".